src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1241,7 +1241,7 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
       def expectedStr = err.expectedTypeStr(pt)
       resolveOverloaded(alts, pt) match {
```

Comment:
I wonder whether `resolveOverloaded` could do `adaptByResult`'s work from the start? What's the rationale for doing this afterwards?